### PR TITLE
Add support for SSA Bold Italic style

### PR DESCRIPTION
**Pull request by abeljim**
@icbaker @szaboa
This is initial PR for Bold and Italic SSA style. I created a SSA file with most of the styles but it seems like VLC has a limit to the number of styles so I wasn't able to have them all. I created a new entry for it in media.exolist.json. Tell me if missed something. Thanks.

### CODE REVIEW

**Comment by abeljim** [comment]
@googlebot I signed it!

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java
Comment (szaboa) [inline library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java]: Based on the specs, only -1 means true, or am I missing something? Also you could simplify it by inlining the return like `return Integer.parseInt(bold) == -1;`. Same for `parseItalic`.

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java
Comment (szaboa) [inline library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java]: I wouldn't put the `else` in new lines, to be consistent with the rest of the code. Also if we want to make it really consistent I'd add these leading comments `/* start= */` and `/* end= */` at the start of the 2nd and 3rd params of `setSpan` - similarly how it is done for the primary color span.

And lastly, you have a couple of missing whitespaces, e.g. `if(style.bold){` -> `if (style.bold) {`

**Review by szaboa — COMMENTED** [review]
Thanks for this, did a quick review, but it all depends on @icbaker ;)

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java
Comment (abeljim) [reply library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java]: Yeah based on the specs it should only be -1 but when I was testing with vlc they took 1 as well. Do you think we should go for a clean implementation or a more forgiving one?

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java
Comment (abeljim) [reply library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java]: Thank you. I will fix this.

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java
Comment (szaboa) [reply library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java]: I've also checked the libass implementation and it seems they are doing the same (-1, 1): https://github.com/libass/libass/blob/82b225b3d6653091d028b39d561d185ed76a7be5/libass/ass_parse.c#L118

So IMO we can go with the more forgiving implementation, but lets wait for @icbaker 's confirmation.

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java
Comment (icbaker) [reply library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaDecoder.java]: I agree with @szaboa - but don't worry too much about the styling/whitespace. When we merge PRs we run them through https://github.com/google/google-java-format (same for all the code we author ourselves), which sorts all this stuff out without needing to think too hard about it.

File: library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java
Comment (icbaker) [reply library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java]: +1 to more forgiving

though weirdly for italic it seems libass **only** consider 1 to mean `true` (just below the line @szaboa linked), while the spec only allows `-1`...? P

Maybe it would be easier just to do `!= 0`?

File: demos/main/src/main/assets/media.exolist.json
Comment (icbaker) [inline demos/main/src/main/assets/media.exolist.json:515]: Thanks for including this to make testing easier, it's really helpful - FYI I'll drop it before actually merging the PR though.

**Review by icbaker — APPROVED** [review]
Looks good, thanks for the contribution! I'll work on getting this merged

**Comment by icbaker** [comment]
> Looks good, thanks for the contribution! I'll work on getting this merged

Correction: I'll wait to see if you decide to use `!= 0` and then merge it :)

**Comment by abeljim** [comment]
I think we should stick with 1 and -1 for now. This way defaults to false when input is unclear. This also follows the libass way. I will later go through and make a file with actually styles from content to have another reference.

File: demos/main/src/main/assets/media.exolist.json
Comment (icbaker) [reply demos/main/src/main/assets/media.exolist.json:515]: Ah I didn't realise this exercises more styles than just italic/bold - this does actually look like a useful file to add to the demo app (but probably not as part of this PR).

If you're happy to contribute this file, can you open a separate PR containing just the SSA file and I'll take it from there an upload it to our cloud bucket and add an entry to the demo app. Thanks!

**Pull request merged**
